pkg/tuf/client.go

```diff
@@ -677,10 +677,12 @@ func (d *diskCache) Set(p string, b []byte) error {
 	if err := d.memory.Set(p, b); err != nil {
 		return err
 	}
-	if err := os.MkdirAll(d.base, 0o700); err != nil {
+
+	fp := filepath.FromSlash(filepath.Join(d.base, p))
+	if err := os.MkdirAll(filepath.Dir(fp), 0o700); err != nil {
```

Comment:
Silly question, what if the file exists but with different permissions?

Reply:
I think that's a valid cause of "your cache is corrupt" then, if you were manipulating it in a weird way.